extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/pipeline/HttpDataSource.java

```diff
@@ -46,21 +45,22 @@ public Stream<Part> openPartStream() {
     private HttpPart getPart() {
         var request = requestFactory.toRequest(params);
         monitor.debug(() -> "HttpDataSource sends request: " + request.toString());
-        try (var response = httpClient.execute(request)) {
-            var body = response.body();
-            var stringBody = body != null ? body.string() : null;
-            if (stringBody == null) {
-                throw new EdcException(format("Received empty response body transferring HTTP data for request %s: %s", requestId, response.code()));
-            }
+        try  {
+            // NB: Do not close the response as the body input stream needs to be read after this method returns. The response closes the body stream.
```

Comment:
When there is a null body, the response shouldn't be closed (it will throw an exception). I'll put a close attempt in the case of a failed request for the case where there is a body returned, but that will need to be wrapped in a try block to catch the case where the body is null.